src/lib.rs

```diff
@@ -145,6 +145,9 @@ pub fn read_choice(
         }
         match ans.parse::<usize>() {
             Ok(a) => {
+                if a < 1 {
+                    println!("{} is not a valid option (too small)", a);
```

Comment:
This will print the error messages, and then panic anyway. To avoid this you need to add a `continue` statement after you've printed this message, to start the data collection again.

Reply:
I've done this for you so I can merge. :)

Reply:
Yes 🤦 I made that change locally and forgot to add that in the PR! Good catch 👍